### PR TITLE
Fixed issue " UI text size does not match theme "

### PR DESCRIPTION
**Pull request by Ayush-singla27**
Fixed issue #69781.  Now theme preview have font size as 16 and font as OpenSans_SemiBold which is same the scene button.
Screenshot 2022-12-20 021636

### CODE REVIEW

File: editor/editor_fonts.cpp [hunk 1]
```diff
@@ -226,6 +227,23 @@ void editor_register_fonts(Ref<Theme> p_theme) {
 	default_fc->set_spacing(TextServer::SPACING_TOP, -EDSCALE);
 	default_fc->set_spacing(TextServer::SPACING_BOTTOM, -EDSCALE);
 
+	Ref<FontVariation> default_theme_fc;
+	default_theme_fc.instantiate();
+	if (custom_font_path.length() > 0 && dir->file_exists(custom_font_path)) {
```
Comment (KoBeWi) [inline editor/editor_fonts.cpp:232]:
```suggestion
	if (!custom_font_path.is_empty() && dir->file_exists(custom_font_path)) {
```

**Comment by KoBeWi** [comment]
I'm a bit puzzled why is this needed. The font is already defined in the default theme, as evident by its appearance in the editor's viewport, so not sure why the theme editor would appear differently. Is it using a wrong fallback for some reason?

File: editor/editor_fonts.cpp (hunk 1 above)
Comment (Ayush-singla27) [reply editor/editor_fonts.cpp:232]: doesn't `is_empty()` check for `length() >0` ?

Comment (KoBeWi) [reply editor/editor_fonts.cpp:232]: Yes, we just prefer it by convention over checking size.

Comment (Ayush-singla27) [reply editor/editor_fonts.cpp:232]: ohk

**Comment by KoBeWi** [comment]
> As usual, when you nest controls inside of the editor, the editor theme leaks.

Can't it be fixed by assigning the default theme somewhere or adding some intermediate node?

**Comment by YuriSizov** [comment]
Default theme doesn't always override things, some things are left to fallback into their respective default values, without any theme definitions. A dummy control would break the layout.

My plan is to have a way to define contexts for specific branches so that the project theme does not leak into the editor UI and the editor theme doesn't leak into various previews. It'll be great, but in 4.1.

**Comment by KoBeWi** [comment]
> A dummy control would break the layout.

It's possible to position the "root" node manually. You just need to place it where it would appear in the container (could even copy position from non-dummy node) and child controls will be laid automatically.

**Comment by YuriSizov** [comment]
Of course you can, you just need to do all the positioning and container logic manually, reporting it back and forth. Is it worth it for this? If we need a temp fix, we can figure something less error prone as a hack (an empty theme that enforces only the default font value for the theme preview).

And I can fix it properly in 4.1 with my planned solution.

**Comment by YuriSizov** [comment]
Thanks for your contribution! I've opened a new PR to correctly address the issue https://github.com/godotengine/godot/pull/70404.

**Pull request closed without merging**